unstructured/partition/lang.py

```diff
@@ -234,12 +234,13 @@ def detect_languages(
     # For example, partition_msg relies on partition_html and partition_text, but the metadata
     # gets overwritten after elements have been returned by _html and _text,
     # so `languages` would be detected twice.
-    if languages[0] == "":
-        return [""]
+    # Also return None if there is no text.
+    if languages[0] == "" or text.strip == "":
+        return None
 
-    # Default to "eng" if text is empty or it has only ascii characters and is short
-    if text.strip() == "" or (re.match(r"^[\x00-\x7F]+$", text) and len(text) < 20):
-        return ["eng"]  # english as default
+    # Default to "eng" if text has only ascii characters and is short
+    if re.match(r"^[\x00-\x7F]+$", text) and len(text) < 20:
+        return ["eng"]
```

Comment:
Should this default also be None? Since the short pieces of text all being classified as English is what prompted this removal of the default in the first place.

Reply:
I guess I misunderstood then. It can be, but in that case it should probably just be simplified to returning `None` if the text isn't long enough to reliably detect the language. It'd be worth investigating what that minimum is, though, if there is one for reliably detecting languages with langdetect, but I suspect that also greatly depends on the actual words that are being used in the tests--doesn't seem worthwhile for us to try and come up with those tests ourselves, but if we can find some existing info on it, that'd be nice.

As it's written now, it assumes the following:
 - the vast majority of short text is English
 - it's preferred to have some short text _incorrectly_ identified as English than to have all short text have no language detected

Changing it assumes:
 - users prefer not having language detected more than incorrect language detection
 - users are okay with language not being detected for all elements under a certain length (or we can just get rid of this snippet entirely and solely rely on what langdetect returns, but we've seen it return 3+ languages for short English text)
 
 I'm fine with either decision.

Reply:
Thanks for outlining the tradeoff. In regards to
> it's preferred to have some short text incorrectly identified as English than to have all short text have no language detected
I'm not sure that that would actually be preferred, because in the case of short non-English text (which is frequently occurring in unstructured element text outputs), labeling it as English will feel glaringly wrong.

> users prefer not having language detected more than incorrect language detection
While we figure out the most accurate way to perform language detection, I'm leaning removing any language being assigned to short elements. Maybe we decrease the character threshold to 10, or switch to a word count of 6?
I agree with you that we shouldn't solely rely on what langdetect returns though

Reply:
Re slack:
"...I know we often have very short element text in unstructured output, so there may barely be any language detection / the impact of incorrectly identified English text will be much more visible. for ex. if a document is in Korean and every element is 3-5 characters, all of them getting labeled as English will feel really wrong"

I already handled that before with the regex pattern. It matches for ascii characters, so any text that has text other than ascii characters will not default to English.

Reply:
@qued Do you have any advice/input on this?

Reply:
To clarify the point about ascii characters, if an element has text that contains special characters (like ñ, å, or Korean/Mandarin/etc.) it will _not_ default to English

Reply:
Ok, I see what you mean about the ascii character check - if a language uses non-ascii characters (accented letters or just entirely separate charsets), it won't count as English, which removes many of the common languages that could result in short text. So 

> short text incorrectly identified as English

is actually a very rare case? 
In that case, I am fine with leaving this as default English. If we can find some cases of this incorrectly labeling text as English, let's add those test cases for visibility of the problem.

Reply:
> Maybe we decrease the character threshold to 10, or switch to a word count of 6? I agree with you that we shouldn't solely rely on what langdetect returns though

I think switching to word count instead of characters is a good idea if we keep this snippet

Reply:
> is actually a very rare case?
> In that case, I am fine with leaving this as default English. If we can find some cases of this incorrectly labeling text as English, let's add those test cases for visibility of the problem.

Yes, should be rather rare and only happen if a snippet of another language only has ascii characters. Ex "Die deutsche Sprache bzw. Deutsch ist eine westgermanische Sprache."